src/lib/parser/core/PokemonSaveParser.ts

```diff
@@ -73,6 +73,7 @@ export class PokemonSaveParser {
   // Memory mode properties
   private webSocketClient: MgbaWebSocketClient | null = null
   private isMemoryMode = false
+  private currentPartyPokemon: PokemonBase[] = []
```

Comment:
we should refactor the logic to not need this but still be concise, simple and well structured